pipelinewise/cli/pipelinewise.py

```
@@ -1621,6 +1655,141 @@ def encrypt_string(self):
+        log_dir = self.get_tap_log_dir(target_id, tap_id)
+        current_time = datetime.utcnow().strftime('%Y%m%d_%H%M%S')
+
+        try:
+            with pidfile.PIDFile(self.tap['files']['pidfile']):
+                self.tap_run_log_file = os.path.join(
+                    log_dir, f'{target_id}-{tap_id}-{current_time}.partialsync.log'
+                )
+
+                # Create parameters as NamedTuples
+                tap_params = TapParams(
+                    tap_id=tap_id,
+                    type=tap_type,
+                    bin=self.tap_bin,
+                    python_bin=self.tap_python_bin,
+                    config=tap_config,
+                    properties=tap_properties,
+                    state=tap_state,
+                )
+
+                target_params = TargetParams(
+                    target_id=target_id,
+                    type=target_type,
+                    bin=self.target_bin,
+                    python_bin=self.target_python_bin,
+                    config=cons_target_config,
+                )
+
+                transform_params = TransformParams(
+                    bin=self.transform_field_bin,
+                    config=tap_transformation,
+                    python_bin=self.transform_field_python_bin,
+                    tap_id=tap_id,
+                    target_id=target_id,
+                )
+
+                self.run_tap_partialsync(
+                    tap=tap_params, target=target_params, transform=transform_params,
+                )
+
+        except pidfile.AlreadyRunningError:
+            self.logger.error('Another instance of the tap is already running.')
+            sys.exit(1)
+        # Delete temp file if there is any
+        except commands.RunCommandException as exc:
+            self.logger.exception(exc)
+            self.send_alert(message=f'Failed to sync tables in {tap_id} tap', exc=exc)
+            sys.exit(1)
```

Please raise an exception

---

- [ ] Fix it!

---

same as above

---

- [ ] Raise Exception